Document Named Parameters in java native client (#1794)
ClickHouse/clickhouse-docs: docs/en/integrations/language-clients/java/index.md

List<ClickHouseResponseSummary> results = future.get();
```

#### Named Parameters

> As being shown in the previous examples, you can pass parameters by name rather than relying solely on their position in the parameter list.
> This capability is available using `params` function.

Reviewer: ```suggestion
You can pass parameters by name rather than relying solely on their position in the parameter list. This capability is available using `params` function.
```

```java
ClickHouseResponse response = client.connect(endpoint) // or client.connect(endpoints)
    // you'll have to parse response manually if using a different format
    .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
    .query("select * from my_table where name=:name limit :limit")
    .params("Ben", 1000).executeAndWait()) {
    ...
    // Additional `params` signatures allow you to pass different Java types such as Strings, Objects, Arrays, and more.

> 

Reviewer: ```suggestion
```

```

> :::note
> All params signatures involving `String` type (`String`, `String[]`, `Map<String, String>`) assumes the **keys being passed are
> valid ClickHouse SQL strings**.
> For instance
> 
> ```java
> ClickHouseResponse response = client.connect(endpoint) // or client.connect(endpoints)
>     // you'll have to parse response manually if using a different format
>     .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
>     .query("select * from my_table where name=:name")
>     .params(Map.of("name","'Ben'")).executeAndWait()) {
>     ...
>    ```
> 
> If you prefer not to parse String objects to ClickHouse SQL manually,
> you can use the helper function `ClickHouseValues.convertToSqlExpression` located at `com.clickhouse.data`:
> 
> ```java
> 
> ClickHouseResponse response = client.connect(endpoint) // or client.connect(endpoints)
>     // you'll have to parse response manually if using a different format
>     .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
>     .query("select * from my_table where name=:name")
>     .params(Map.of("name", ClickHouseValues.convertToSqlExpression("Ben's"))).executeAndWait()) {
>     ...
>    ```
> 
> For the example above, `ClickHouseValues.convertToSqlExpression` will escape the inner single quote, and surround the
> variable with a valid single quotes.
> 
> Other types, such as `Integer`, `UUID`, `Array` and even `Enum` will be converted automatically inside `params`.
> :::

Reviewer: ````suggestion
:::note
All `params` signatures involving `String` type (`String`, `String[]`, `Map<String, String>`) assume the keys being passed are valid ClickHouse SQL strings. For instance:

```java
ClickHouseResponse response = client.connect(endpoint) // or client.connect(endpoints)
    // You'll have to parse response manually if using a different format.
    .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
    .query("select * from my_table where name=:name")
    .params(Map.of("name","'Ben'")).executeAndWait()) {
    ...
```

If you prefer not to parse String objects to ClickHouse SQL manually, you can use the helper function `ClickHouseValues.convertToSqlExpression` located at `com.clickhouse.data`:

```java
ClickHouseResponse response = client.connect(endpoint) // or client.connect(endpoints)
    // You'll have to parse response manually if using a different format
    .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
    .query("select * from my_table where name=:name")
    .params(Map.of("name", ClickHouseValues.convertToSqlExpression("Ben's"))).executeAndWait()) {
    ...
```

In the example above, `ClickHouseValues.convertToSqlExpression` will escape the inner single quote, and surround the variable with a valid single quotes.

Other types, such as `Integer`, `UUID`, `Array` and `Enum` will be converted automatically inside `params`.
:::
```
